src/robot/libraries/Process.py

```diff
@@ -524,8 +526,37 @@ def _manage_process_timeout(self, handle, on_timeout):
 
     def _wait(self, process):
         result = self._results[process]
-        result.rc = process.wait() or 0
-        result.close_streams()
+
+        _take_stdout = (process.stdout and (not process.stdout.closed))
+        _take_stderr = (process.stderr and (not process.stderr.closed))
+
+        # This is needed to handle stdin closes identically to how it was when the
+        # wait function was used.
+        _stdin = process.stdin
+        process.stdin = process.stdin if (process.stdin and not process.stdin.closed) else io.StringIO("")
+        while True:
+            with contextlib.suppress(subprocess.TimeoutExpired):
+                # this is to allow the process to be aborted when on windows
+                # an exception is used to mange the test timeout situation
+                (_stdout, _stderr,) = process.communicate(timeout=0.1)
+                break
```

Comment:
This isn't related to #4173 anymore and should rather be submitted as a separate PR.

In my opinion `try/except` would make the code easier to understand that `contextlib.suppress`.